ruleguard/logwrappederr.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package ruleguard
+
+import (
+	"github.com/quasilyte/go-ruleguard/dsl"
+)
+
+// logwrappederr finds instances of Sync Gateway logging where we are attempting to use %w on an error. Should use %s or %v.
+func logwrappederr(m dsl.Matcher) {
+	m.
+		Match(
+			// this is horrible but the variadics after $err don't match with the optional wildcards $*_
+			// also no way to regex match to provide non-package qualified versions
+			`base.PanicfCtx($ctx, $format, $err)`,
```

Comment:
I think you can do `base.PanicfCtx($ctx, $format, $*args)` and go vet will test if you use %w without an err field

Reply:
Reworked this rule to be a bit less strict and simpler (it'll detect ALL usages of `%w` no matter the type of args)